File: python/misc/maximum_edge_of_a_triangle.py
#!/usr/bin/env python3

"""Maximum Edge of a Triangle.

Create a function that finds the maximum range of a triangles third edge.

Source:
https://edabit.com/challenge/Zerwo2AENbvRZTe83
"""


def next_edge(side_1: int, side_2: int) -> int:
    """Find the maximum range of a triangles third edge."""
    return (side_1 + side_2) - 1


def main():
    """Run sample next_edge. Do not import."""
    assert next_edge(5, 4) == 8
    assert next_edge(8, 3) == 10
    assert next_edge(7, 9) == 15
    assert next_edge(10, 4) == 13
    assert next_edge(7, 2) == 8
    print('Passed.')


if __name__ == "__main__":
    main()

File: javascript/edabit/maximum_edge_of_a_triangle.js
/*
# Maximum Edge of a Triangle

Create a function that finds the maximum range of a triangle 's third edge.

## Examples
* nextEdge(8, 10)➞ 17
* nextEdge(5, 7)➞ 11
* nextEdge(9, 2)➞ 10

## Notes
(side1 + side2) - 1 = maximum range of third edge.
All given triangles have side lengths that are positive integers.
Don 't forget to return the result.

Source:
https: //edabit.com/challenge/nhXofMMyrowMyr9Nv
*/
function nextEdge(side1, side2) {
    return (side1 + side2) - 1;
}


/**
 * Run sample nextEdge functions, print results to console.log.
 */
console.log(nextEdge(5, 4) == 8)
console.log(nextEdge(8, 3) == 10)
console.log(nextEdge(7, 9) == 15)
console.log(nextEdge(10, 4) == 13)
console.log(nextEdge(7, 2) == 8)